lib/pages/AccountCreation.dart

```diff
@@ -0,0 +1,170 @@
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          'Account Sign up'
+        ),
+        centerTitle: true,
+        backgroundColor: Colors.amber,
+      ),
+      body: SafeArea(
+        child: Form(
+          key: _accFormKey,
+          autovalidate: true,
+          child: Column(
+            children: <Widget>[
+              new InputEmail(account: account),
+              new InputName(account: account),
+              new InputPassword(account: account),
+              new SubmitButton(accFormKey: _accFormKey, account: account)
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class SubmitButton extends StatelessWidget {
+  const SubmitButton({
+    Key key,
+    @required GlobalKey<FormState> accFormKey,
+    @required this.account,
+  }) : _accFormKey = accFormKey, super(key: key);
+
+  final GlobalKey<FormState> _accFormKey;
+  final AccountService account;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(left: 40.0, right: 40.0, top: 5.0),
+      child: RaisedButton(
+        onPressed: () {
+          if(_accFormKey.currentState.validate()) {
+            _accFormKey.currentState.save();
+
+            print(this.account.name);
+            print(this.account.email);
+            print(this.account.password);
+            account.saveAccount();
+          }
+        },
+        color: Colors.amber,
+        child: Text('Sign Up'),
+      ),
+    );
+  }
+}
+
+class InputPassword extends StatelessWidget {
+  const InputPassword({
+    Key key,
+    @required this.account,
+  }) : super(key: key);
+
+  final AccountService account;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(left: 40.0, right: 40.0, top: 5.0),
+      child: TextFormField(
+        obscureText: true,
+        decoration: const InputDecoration(
+          focusedBorder: OutlineInputBorder(
+            borderSide: BorderSide(color: Colors.grey, width: 1.0)
+          ),
+          enabledBorder: OutlineInputBorder(
+            borderSide: BorderSide(color: Colors.grey, width: 1.0)
+          ),
+          labelText: 'PASSWORD',
+        ),
+        validator: (value) {
+          if(value.length < 8)
+            return 'Invalid password, minimum length: 8 characters';
+          else if(value.length > 16)
+            return 'Invalid password, maximum length 16 characters';
+          else
+            return null;
+        },
+        onSaved: (value) => account.password = value,
+      ),
+    );
+  }
+}
+
+class InputName extends StatelessWidget {
+  const InputName({
+    Key key,
+    @required this.account,
+  }) : super(key: key);
+
+  final AccountService account;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(left: 40.0, right: 40.0, top: 5.0),
+      child: TextFormField(
+        decoration: const InputDecoration(
```

Comment:
Eventually even this decorators should be implemented by us. Kinda like wrapped in another class

Reply:
We could leave for future, second phase.